pkg/healthcheck/healthcheck_test.go

```diff
@@ -1645,6 +1645,65 @@ func TestValidateDataPlanePods(t *testing.T) {
 			t.Fatalf("Unexpected error: %s", err)
 		}
 	})
+
+	// This test relates to https://github.com/linkerd/linkerd2/issues/6128
+	t.Run("Returns nil if some pods are in the Succeeded phase and their proxies are no longer ready", func(t *testing.T) {
+		pods := []corev1.Pod{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: "emoji-d9c7866bb-7v74n"},
+				Status: corev1.PodStatus{
+					Phase: "Succeeded",
```

Comment:
Do we need to specify `Reason: "Completed"` here? IIUC a pod's status is `completed` when its phase is `Succeeded` and the reason is `Completed`, it seems the test pass without it though so not sure but would love to hear more about it, pod statuses are still a bit foreign to me.

Reply:
Thanks for pointing this out. It looks like this test isn't quite right so I'll be fixing that in the next commit. However, the test can still fail if we don't include the `Reason: Completed` field. It definitely wouldn't hurt to include that since that is closer to what we'd get in an actual environment.